### PR TITLE
DOCS: Add mapping guide to style docs

### LINKED ISSUE #2917: Style Guide Docs -- Mappings section
What was intended for the mappings section of the style guide? It's a TODO currently. If somebody could point me to a few good `mapping` usage examples I'd happily complete this section of `style-guide.rst`.

### PR DESCRIPTION
**Pull request by ChrisChinchilla**
### Description

Thanks to @aupiff for the contribution this PR closes https://github.com/ethereum/solidity/issues/2917 by adding mapping details to style guide.

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [x] New tests have been created which fail without the change (if possible)
- [x] README / documentation was extended, if necessary
- [x] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

### CODE REVIEW

File: docs/style-guide.rst [hunk 1]
```diff
@@ -809,7 +809,21 @@ possible permutations for function declarations.
 Mappings
 ========
 
-TODO
+Surround the type of a ``mapping`` by parentheses. In variable declarations, separate the keyword ``mapping`` from its type by a single space. Do not separate any nested ``mapping`` keyword from its type by whitespace.
```
Comment (chriseth) [inline docs/style-guide.rst]: The parentheses are part of the syntax, they are not style.

Comment (chriseth) [inline docs/style-guide.rst]: Is it really standard practice to change the spacing for nested mappings?

Comment (ChrisChinchilla) [reply docs/style-guide.rst]: I don't know, I am merely adding the result of the conversation from https://github.com/ethereum/solidity/issues/2917 if there's something different let me know.

**Comment by ChrisChinchilla** [comment]
Updates based on comment here - https://github.com/ethereum/solidity/issues/2917#issuecomment-443242773

**Review by chriseth — DISMISSED** [review]

**Comment by chriseth** [comment]
Rebased.

**Review by chriseth — APPROVED** [review]

**Pull request merged**